### PR TITLE
feat: UpdateIP function on instance API

### PR DESCRIPTION
**Pull request by DenBeke**

### CODE REVIEW

File: api/instance/v1/instance_utils.go [hunk 1]
```diff
@@ -57,3 +57,27 @@ func (s *API) DetachIP(req *DetachIPRequest, opts ...scw.RequestOption) (*Detach
 
 	return &DetachIPResponse{IP: ipResponse.IP}, nil
 }
+
+// UpdateIPRequest contains the parameters to updaten an IP
```
Comment (jerome-quere) [inline api/instance/v1/instance_utils.go]: * update

File: api/instance/v1/instance_utils.go [hunk 2]
```diff
@@ -57,3 +57,27 @@ func (s *API) DetachIP(req *DetachIPRequest, opts ...scw.RequestOption) (*Detach
...
+type UpdateIPRequest struct {
+	Zone    utils.Zone `json:"-"`
+	IPID    string     `json:"-"`
+	Reverse *string    `json:"reverse"`
```
Comment (jerome-quere) [inline api/instance/v1/instance_utils.go]: Add comment that specify:
` If Reverse is an empty string reverse will be remove`

File: api/instance/v1/instance_utils.go [hunk 3]
```diff
@@ -57,3 +57,27 @@ func (s *API) DetachIP(req *DetachIPRequest, opts ...scw.RequestOption) (*Detach
...
+}
+
+// UpdateIP updates an IP
+func (s *API) UpdateIP(req *UpdateIPRequest, opts ...scw.RequestOption) (*UpdateIPResponse, error) {
+	if *req.Reverse == "" {
```
Comment (kindermoumoute) [inline api/instance/v1/instance_utils.go]: if req.Reverse!= nil && *req.Reverse == ""

File: api/instance/v1/instance_utils.go [hunk 4]
```diff
@@ -57,3 +57,28 @@ func (s *API) DetachIP(req *DetachIPRequest, opts ...scw.RequestOption) (*Detach
...
+// if Reverse is an empty string, the reverse will be removed
+type UpdateIPRequest struct {
+	Zone    utils.Zone `json:"-"`
+	IPID    string     `json:"-"`
+	Reverse *string    `json:"reverse"`
+}
+
+// UpdateIP updates an IP
+func (s *API) UpdateIP(req *UpdateIPRequest, opts ...scw.RequestOption) (*UpdateIPResponse, error) {
+	if req.Reverse != nil && *req.Reverse == "" {
```
Comment (kindermoumoute) [inline api/instance/v1/instance_utils.go]: You should handle when reverse is nil:
```
var reverse **string
if req.Reverse != nil  && *req.Reverse == "" {
req.Reverse = nil
reverse = &req.Reverse
}
	ipResponse, err := s.updateIP(&updateIPRequest{
		Zone:    req.Zone,
		IPID:    req.IPID,
		Reverse: reverse,
	})
```

**Pull request merged**